server/inputs/registration.ts

```diff
@@ -0,0 +1,73 @@
+/// <reference path="../../apis/registration.d.ts" />
+import * as request from "request-promise-native";
+
+interface IRegistrationOpts {
+	url: string;
+	key: string;
+}
+
+export class Registration {
+	url: string;
+	key: string;
+
+	constructor(opts: IRegistrationOpts) {
+		this.url = opts.url;
+		this.key = new Buffer(opts.key).toString("base64");
+	}
+
+	async user(id: string, selection_set: string[]) {
+		const vars = selection_set.map((_, i) => `$select_${i}`).join(" ");
+		const var_map: {[selection: string]: string} = {};
+
+		selection_set.forEach((selection, i, _) => {
+			var_map[`$select_${i}`] = selection;
+		}, {});
+
+		const result = await this.query(`{
+			user(id: $id) {
+				${vars}
+			}
+		}`, var_map);
+		return result.user;
+	}
```

Comment:
I get a graphql syntax error when I try to call the `user` method. 

This was the error:
```
{ StatusCodeError: 400 - {"errors":[{"message":"Syntax Error GraphQL request (3:5) Expected Name, found $\n\n2: \t\t\tuser(id: $id) {\n3: \t\t\t\t$select_0 $select_1\n       ^\n4: \t\t\t}\n","locations":[{"line":3,"column":5}]}]}
```

It works when I change the method to this:
```
	async user(id: string, selection_set: string[]) {
		let vars: string = selection_set.join(' ');

		const result = await this.query(`{
			user(id: "${id}") {
				${vars}
			}
		}`);
		return result.user;
	}
```

Reply:
:frowning_face: I think maybe `$vars` shouldn't have a `$`, technically you want to use the vars map so it escapes special characters, but you can change it for now and file a bug.

Reply:
Yeah, I had trouble querying with the vars map, but I'm not too familiar with graphql.